### PR TITLE
Cleanup some issues with exec

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

File: daemon/container.go [hunk 1]
```diff
@@ -790,7 +790,7 @@ func (container *Container) getExecIDs() []string {
 	return container.execCommands.List()
 }
 
-func (container *Container) exec(ExecConfig *ExecConfig) error {
+func (container *Container) exec(ec *ExecConfig) error {
```
Comment (LK4D4) [inline daemon/container.go]: best

**Comment by LK4D4** [comment]
@cpuguy83 Is there possible to write some tests?

File: daemon/exec.go [hunk 2]
```diff
@@ -262,6 +261,9 @@ func (d *Daemon) ContainerExecStart(name string, stdin io.ReadCloser, stdout io.
 		}
 		return nil
 	case err := <-execErr:
+		if err := <-attachErr; err != nil {
+			logrus.Warn(err)
```
Comment (duglin) [inline daemon/exec.go]: If the exec is ok but the attach fails, is it really ok to just silently ignore the attach error?  While we log it to the daemon, the end user won't see it.  I'm wondering if we should halt processing instead.

**Comment by duglin** [comment]
@cpuguy83 is this related to the racey testcase?

**Comment by cpuguy83** [comment]
@duglin No, racey test case is due to raciness in libcontainer.

@LK4D4 Not sure about tests for this. Mainly it's fixing a case where we started sending the error to the client (after the last time I touched this), even though they are already getting the error from the stdio stream, so it gets sent twice in some cases... and the error wasn't properly formatted for StdCopy so it produces some issues in the debug logs from StdCopy, not much to catch here.

File: daemon/exec.go (hunk 2 above)
Comment (cpuguy83) [inline daemon/exec.go]: Really if exec didn't fail I think this should always be nil, it's just waiting for the attach to finish before continuing.

**Comment by LK4D4** [comment]
@cpuguy83 maybe it's okay without tests
LGTM

File: daemon/exec.go (hunk 2 above)
Comment (duglin) [inline daemon/exec.go]: but what if it isn't nil?

Comment (cpuguy83) [inline daemon/exec.go]: Updated to return the attach error if the exec error is nil.

**Comment by duglin** [comment]
@cpuguy83 thanks for the update. Testing/reviewing now....

**Comment by duglin** [comment]
LGTM
But I'd prefer one more set of eyes to review it before we merge.

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**